kratos/processes/find_intersected_geometrical_objects_process.cpp

```diff
@@ -8,248 +8,476 @@
+
+std::vector<PointerVector<GeometricalObject>>& FindIntersectedGeometricalObjectsProcess::GetIntersections()
+{
+    return mIntersectedObjects;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+ModelPart& FindIntersectedGeometricalObjectsProcess::GetModelPart1()
+{
+    return mrModelPartIntersected;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+ModelPart& FindIntersectedGeometricalObjectsProcess::GetModelPart2()
+{
+    return mrModelPartIntersecting;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+OctreeBinary<OctreeBinaryCell<typename FindIntersectedGeometricalObjectsProcess::ConfigurationType>>* FindIntersectedGeometricalObjectsProcess::GetOctreePointer()
+{
+    return& mOctree;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FindIntersectedGeometricalObjectsProcess::Clear()
+{
+    mIntersectedObjects.clear();
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FindIntersectedGeometricalObjectsProcess::Execute()
+{
+    // Calling initialize first (initialize Octree)
+    ExecuteInitialize();
+
+    OtreeCellVectorType leaves;
+
+    // Iterate over elements
+    auto& r_elements_array = mrModelPartIntersected.Elements();
+    const SizeType number_of_elements = r_elements_array.size();
+
+    const auto it_elements_begin = r_elements_array.begin();
+
+    #pragma omp parallel for private(leaves)
+    for (int i = 0; i < static_cast<int>(number_of_elements); i++) {
+        auto it_elem = it_elements_begin + i;
+        leaves.clear();
+        IdentifyNearEntitiesAndCheckEntityForIntersection(*(it_elem.base()), leaves);
     }
 
-    /// Print information about this object.
-    void FindIntersectedGeometricalObjectsProcess::PrintInfo(std::ostream& rOStream) const {
-        rOStream << Info();
+    // Iterate over conditions
+    auto& r_conditions_array = mrModelPartIntersected.Conditions();
+    const SizeType number_of_conditions = r_conditions_array.size();
+
+    const auto it_conditions_begin = r_conditions_array.begin();
+
+    #pragma omp parallel for private(leaves)
+    for (int i = 0; i < static_cast<int>(number_of_conditions); i++) {
+        auto it_cond = it_conditions_begin + i;
+        leaves.clear();
+        IdentifyNearEntitiesAndCheckEntityForIntersection(*(it_cond.base()), leaves);
     }
```

Comment:
Now you create the Octree for both the elements and conditions of the intersected model part? Does this work if such intersected model part has both types of entities? If it does, doesn't this introduce an overhead? Maybe @pooyan-dadvand  can better clarify this.

Reply:
You cannot if you want to be able to assign flags, because the GeometricalObject does not have flags

Reply:
And the idea of the process is to assign the flag SELECTED

Reply:
You mean that now it is needed to create the leaves of both elements and conditions because the `GeometricalObject` can't handle flags. Is this true?